### ISSUE TITLE
Label clipping at tile borders

### ISSUE THREAD

**Issue by pbnsilva** [mapbox/mapbox-gl-native#15805]
**Platform:** `node`

**Mapbox SDK version:** `v5.0.0`

### Steps to trigger behavior

See below for the code that generates 3 tiles where the problem can be seen. I am using this [style](https://drive.google.com/file/d/1zW3a5AF-Amb1w8n7PLi4VZwq5_m30jzn/view?usp=sharing).

Note that I am using `tile` mode and `symbol-avoid-edges: true`.

This happens very frequently.

### Expected behavior

Label extends across tiles, or label is not shown.

### Actual behavior

![image](https://user-images.githubusercontent.com/2334434/66698972-2d1b3580-ece3-11e9-97e7-cea7b2d587a1.png)

### Code
```javascript
const mbgl = require('@mapbox/mapbox-gl-native')
const request = require('request')
const utils = require('./utils')
const config = require('./config')
const sharp = require('sharp')
const readline = require('readline')
const fs = require('fs')

var tile1 = { 
  zoom: 10.8,
  pxWidth: 563,
  pxHeight: 563,
  width: 513,
  height: 513,
  center: [ 8.371469839463089, 53.59297564603976 ],
  bearing: 0,
  pitch: 0,
  ratio: 1.097485380116959
}

var tile2 = { 
  zoom: 10.8,
  pxWidth: 563,
  pxHeight: 563,
  width: 513,
  height: 513,
  center: [ 8.573706750983263, 53.59297564603976 ],
  bearing: 0,
  pitch: 0,
  ratio: 1.097485380116959
}

var tile3 = {
  zoom: 10.8,
  pxWidth: 563,
  pxHeight: 563,
  width: 513,
  height: 513,
  center: [ 8.77594366250338, 53.59297564603976 ],
  bearing: 0,
  pitch: 0,
  ratio: 1.097485380116959
}

renderTile('tile1', tile1)
renderTile('tile2', tile2)
renderTile('tile3', tile3)

function mapRequest(req, callback) {
  var url = req.url
  if (url.indexOf('access_token') === -1) {
    var urlObject = utils.parseURL(url)
    url = utils.makeAPIURL(urlObject, config.ACCESS_TOKEN)
  }

  makeRequest(url, callback)
}

function makeRequest(url, callback) {
  request({
    url: url,
    encoding: null,
    gzip: true
  }, function (err, res, body) {
    if (err) {
      callback(err)
    } else if (res.statusCode == 200) {
      var response = {}

      if (res.headers.modified) { response.modified = new Date(res.headers.modified) }
      if (res.headers.expires) { response.expires = new Date(res.headers.expires) }
      if (res.headers.etag) { response.etag = res.headers.etag }

      response.data = body

      callback(null, response)
    } else {
      callback(new Error(JSON.parse(body).message))
    }
  })
}

function renderTile(name, tile) {
  var options = {
    request: mapRequest,
    ratio: tile.ratio,
    mode: 'tile'
  }
  var map = new mbgl.Map(options)

  map.load(require('./style.json'))

  map.render({
    zoom: tile.zoom,
    width: tile.width,
    height:  tile.height,
    center: tile.center,
    bearing: 0,
    pitch: 0,
  }, function(err, buffer) {
      if (err) {
        throw err
      }

      map.release()

      var image = sharp(buffer, {
          raw: {
              width: tile.pxWidth,
              height: tile.pxHeight,
              channels: 4
          }
      })

      image.toFile(name + '.png', function(err) {
          if (err) {
            throw err
          }
      })
  })
}
```

**Comment by chloekraw** [comment]
cc @springmeyer

**Comment by pbnsilva** [comment]
This is fixed with `v5.0.2`. Thanks for your support!
